src/CommandLineUtils/Conventions/SubcommandAttributeConvention.cs

```diff
@@ -67,9 +67,13 @@ private void AssertSubcommandIsNotCycled(Type modelType, CommandLineApplication?
         private void AddSubcommandImpl<TSubCommand>(ConventionContext context, SubcommandAttribute subcommand)
             where TSubCommand : class
         {
-#pragma warning disable 618
-            context.Application.Command<TSubCommand>(subcommand.Name!, subcommand.Configure);
-#pragma warning restore 618
+            var commandAttribute = typeof(TSubCommand).GetTypeInfo().GetCustomAttributes(typeof(CommandAttribute)).FirstOrDefault() as CommandAttribute;
+            if(commandAttribute == null)
+            {
+                throw new CommandAttributeMissingException(typeof(TSubCommand));
```

Comment:
This isn’t a behavior I think we should change. The Command attribute is currently optional for subcommand types.  Can you find a way to remove the obsolete API without this? I believe it should work if this method is changed to...
```csharp
context.Application.Command<TSubCommand>(null!, subcommand.Configure);
```

Reply:
Not a problem at all. I misunderstood #139 to mean that `[Command]` would now be required. The subcommand's `Configure()` only set up the Name field, which has been removed, so I removed that method also. Looks like passing null to both works just fine (there are checks in the code to handle those cases, seems).